subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/provider/KotlinScriptClassPathProvider.kt

```diff
@@ -161,9 +166,22 @@ class KotlinScriptClassPathProvider(
     private
     fun produceFrom(id: String, generate: JarGeneratorWithProgress): File =
         jarCache(id) { outputFile ->
-            progressMonitorFor(outputFile, 3).use { progressMonitor ->
-                generateAtomically(outputFile) { generate(it, progressMonitor::onProgress) }
-            }
+            buildOperationExecutor.run(object : RunnableBuildOperation {
```

Comment:
❓ Do we need both? The progress monitor and the build operation?

Reply:
afaik yes, @oehme can you confirm?